padl/transforms.py

```diff
@@ -1847,6 +2009,11 @@ def _pd_splits(self, input_components=0) -> Tuple[Union[int, List],
                 res.append(split)
         final_splits = res
 
+        if self._pd_name is not None:
+            for i, s in enumerate(final_splits):
+                if not isinstance(s, Identity):
+                    final_splits[i] = s - self._pd_name
+
```

Comment:
Re-add `final_splits = tuple(final_splits)` here. We want to always return the final_splits as tuples.

Reply:
ok

Reply:
actually I'm checking with @wuhu if this is needed.

Reply:
Do we know something?

Reply:
Not yet

Reply:
I added it and all the tests are passing, so it should be ok

Reply:
However, it looks like the tests are failing for python 3.7